tests/pending/neg/cc-depfun.scala

```diff
@@ -0,0 +1,14 @@
+import language.experimental.captureChecking
+
+// compare with neg-custom-args/captures/depfun.scala, which produces errors
+// but the errors go away if ->{} gets replaced by ->.
+
+trait Cap { def use(): Unit }
+
+def main() = {
+  val f: (io: Cap^) -> () -> Unit =
+    io => () => io.use()  // error
+
+  val g: (Cap^) -> () -> Unit =
+    io => () => io.use()  // error
+}
```

Comment:
This inconsistency can be reproduced before this PR. It seems that this is a result of `expandAbbreviations`:
https://github.com/lampepfl/dotty/blob/9ed9839e533db22c69bdb1e552ef6eeaacecc333/compiler/src/dotty/tools/dotc/cc/Setup.scala#L286-L296